src/DynamoCore/Engine/CodeGeneration/CompilationEventArgs .cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using ProtoCore.AST.AssociativeAST;
+
+namespace Dynamo.Engine.CodeGeneration
+{
+    /// <summary>
+    /// This class contains data about NodeModel compilation when a NodeModel is being 
+    /// compiled to a list of AST nodes.
+    /// </summary>
+    public class CompilationEventArgs : EventArgs
+    {
+        private List<AssociativeNode> additionalAsts = new List<AssociativeNode>();
+
+        /// <summary>
+        /// Creates CompilationEventArgs
+        /// </summary>
+        /// <param name="node">The node being compiled.</param>
+        /// <param name="inputNodes">List of input ast nodes.</param>
+        /// <param name="context">Compilation context.</param>
+        public CompilationEventArgs(Guid node, IEnumerable<AssociativeNode> inputNodes, CompilationContext context)
```

Comment:
I want to be conservative as we're building a new API - can this be internal to start?